### PR TITLE
Adding CIOnly support to PlatformFacts and adding CIOnly attributes to three tests

### LINKED ISSUE #11311: Eliminate UT failures locally
These tests fail when run locally when sync'd to HEAD:

- Test_ExtractionHonorsFileTimestamp()
- ExtractPackageAsync_PreservesZipEntryTimeAsync()
- ExtractPackageAsync_RequireMode_NoMatchInClientAllowList_Error()

### PR DESCRIPTION
**Pull request by aortiz-msft**
## Bug
Fixes: https://github.com/NuGet/Home/issues/11311

Regression? Last working version: N/A

## Description
Skip three Unit Tests when running them locally since:

- Test_ExtractionHonorsFileTimestamp() and ExtractPackageAsync_PreservesZipEntryTimeAsync() require the host to be in a specific timezone. This can probably be fixed in a subsequent PR. Opened https://github.com/NuGet/Home/issues/11312 to track this improvement.
- ExtractPackageAsync_RequireMode_NoMatchInClientAllowList_Error() only works when running on Windows with Admin privileges. In order to be able to skip this test locally, I added a CIOnly parameter to the PlatformFactAttribute.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception: Not adding new product code. As part of these changes, I validated the the three tests above are skipped locally but run on the CI. I validated the change to the PlatformFactAttribute by making sure that existing tests with a PlatformFact continue to run locally as expected.
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Comment by aortiz-msft** [comment]
@heng-liu - Any idea why the PublicAPI:ReviewNeeded label was added by the bot if only test changes are included in this PR?

File: test/TestUtilities/Test.Utility/PlatformXunitAttributes/CIOnly.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Test.Utility
+{
+    /// <summary>
+    /// Enum to specify if a test should only run on CI or not.
+    /// </summary>
+    public enum CIOnly
```
Comment (nkolev92) [inline test/TestUtilities/Test.Utility/PlatformXunitAttributes/CIOnly.cs]: If there's only 2 values, wouldn't we be better off with a bool instead?

Comment (aortiz-msft) [reply test/TestUtilities/Test.Utility/PlatformXunitAttributes/CIOnly.cs]: The intent with CIOnly.Yes and CIOnly.No is immediately clear compared to a bool,  particularly when looking at the tests outside an IDE.

Comment (nkolev92) [reply test/TestUtilities/Test.Utility/PlatformXunitAttributes/CIOnly.cs]: You can always use a named parameter. 

SkipMono is an example: https://github.com/NuGet/NuGet.Client/blob/e412505169a04262bced0416af592a75179a4871/test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformTheoryAttribute.cs#L52

Comment (aortiz-msft) [reply test/TestUtilities/Test.Utility/PlatformXunitAttributes/CIOnly.cs]: I see. Yes, that makes more sense. Thanks @nkolev92 :)

File: test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformFactAttribute.cs [hunk 2]
```diff
@@ -51,18 +59,30 @@ public override string Skip
...
         /// Provide property values to use this attribute.
         /// </summary>
         public PlatformFactAttribute()
         {
         }
 
+        /// <summary>
+        /// Run only on the given platforms
+        /// </summary>
+        public PlatformFactAttribute(CIOnly ciOnly, params string[] platforms)
```
Comment (zivkan) [inline test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformFactAttribute.cs]: Attributes should not have properties also in the constructor. Instead, you use unnamed parameters for constructor parameters, and named variables for properties. For example `[PlatformFactAttribute(Platform.Windows, CIOnly = true)]`.

Therefore, this constructor should not be added.

Comment (aortiz-msft) [reply test/TestUtilities/Test.Utility/PlatformXunitAttributes/PlatformFactAttribute.cs]: Yup, @nkolev92 had the same feedback. New iteration to be sent out soon.

**Review by zivkan — APPROVED** [review]

**Review by nkolev92 — APPROVED** [review]

**Review by heng-liu — APPROVED** [review]

**Review by erdembayar — APPROVED** [review]

**Comment by zivkan** [comment]
Overriding merge policy, because we need the CIOnly property on PlatformFactAttribute to property skip the tests failing on mac.

**Pull request merged**